### ISSUE TITLE
Dashboard tabs do not always update

### ISSUE THREAD

**Issue by dotmacro** [okTurtles/group-income#2395]
### Problem

On `testing`, sometimes the `SupportHistory` and `TodoHistory` tabs don't update when a new period begins. 

These screenshots are from today, October 27th, meaning they're missing one completed period and the current period (ending November 3rd). 

![noDashboardUpdates--support](https://github.com/user-attachments/assets/c1a0bd6d-625e-46e0-af1f-99323ab1a8dc)
![noDashboardUpdates--todo](https://github.com/user-attachments/assets/4285220c-3879-43a5-8a9f-c9a50b6d2ff8)

![noDashboardUpdates--payments](https://github.com/user-attachments/assets/1e2b6b16-176b-4bf0-a4df-5d2d4a1159f7)

~~I haven't been able to pin down a cause because it doesn't affect all groups on `testing`, nor does it affect all groups this user is a member of on `testing`. In case it helps, the affected group has mincome set in Bitcoin and the groupName is emoji. Changing the group name to Latin characters does not fix the issue.~~

After further testing, as @taoeffect suggests, this reliably occurs when no payments have been made. In this case, the dashboard should still be updated. 

### Solution

The Dashboard tabs should show all months since the first income details were entered. 
1. If no payments were made && there was no unmet need, the `SupportHistory` and `TODOHistory` should both be 100% for that period. 
2. If no payments were made but there was unmet need, the `SupportHistory` and `TODOHistory` should both show the proper calculations for missed support and missed TODOs, i.e. 0%, for that period.

**Comment by taoeffect** [comment]
I'm guessing this might happen when there were zero payments for the period.

**Comment by dotmacro** [comment]
You're correct that there were no payments made for either missing period, yes! That does seem to be a cause because making a payment in the current period does add that period to the dashboard tabs. However, this also makes it look as though the current period is the only period in the group's history: 
![image](https://github.com/user-attachments/assets/04c5802a-9d40-42e4-83ce-e2ffec38a4c0)

The app should should show history, even if no payments have yet been made. 

Once the first payment is made, should the dashboard reset to make it look as though the current period is the only period in the group's history?

**Comment by dotmacro** [comment]
After further testing, this reliably occurs when no payments have been made. In this case, the dashboard should still be updated. 

The Dashboard tabs should show all months since the first income details were entered. 
1. If no payments have been made and there was no unmet need, the `SupportHistory` and `TODOHistory` should both be 100%. 
2. If no payments have been made but there were unmet needs, the `SupportHistory` and `TODOHistory` should both show the proper calculations for missed support and missed TODOs, i.e. 0%.

Issue updated.
